test/functional/api/v2/override_values_controller_test.rb

```diff
@@ -67,4 +67,26 @@ class Api::V2::OverrideValuesControllerTest < ActionController::TestCase
     end
     assert_response :success
   end
+
+  test "should not create override value without value" do
+    k = FactoryGirl.create(:lookup_key, :puppetclass => puppetclasses(:two))
+    refute k.override
+    assert_difference('LookupValue.count', 0) do
+      post :create,  {:smart_variable_id => k.id, :override_value => { :value => 'xyz=10' } }
+    end
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert_equal("Validation failed: Match can't be blank", response['error']['message'])
+    assert_response :error
+  end
+
+  test "should not create override value without match" do
+    k = FactoryGirl.create(:lookup_key, :puppetclass => puppetclasses(:two))
```

Comment:
Almost the same preparation code as the other test, can you DRY it a bit? It looks like you can do something like

```ruby 
[{ :value => 'xyz=10'}, { :match => 'string'}].each do |override_value|
  test "should not create override value without #{override_value.key}" do
    lookup_key = FactoryGirl.create(:lookup_key, :puppetclass => puppetclasses(:two))
    refute lookup_key.override
    assert_difference('LookupValue.count', 0) do
      post :create, { :smart_variable_id => lookup_key.id, :override_value => override_value }
    end
    response = ActiveSupport::JSON.decode(@response.body)
    assert_matches /#{override_value.key} can't be blank/, response['error']['message']
    assert_response :error
  end
end
```